nUpdate.Shared/Updating/UpdateManager.cs

```diff
@@ -471,5 +485,49 @@ bool Validate(KeyValuePair<UpdateVersion, string> filePathItem)
             Cleanup();
             return false;
         }
+
+        /// <summary>
+        ///     Convert <see cref="UpdateConfiguration.UpdatePackageUri"/> uri-path to the <see cref="UpdateManager.UpdateConfigurationFileUri"/> uri-path 
+        ///     If <see cref="UseDynamicUpdateUri"/> is false, return the original value
+        /// </summary>
+        /// <param name="updatePackageUri">The update-packages uri. <see cref="UpdateConfiguration.UpdatePackageUri"/></param>
+        /// <param name="literalVersion">The update-packages literalVersion. <see cref="UpdateConfiguration.LiteralVersion"/></param>
+        /// <returns>
+        ///     Returns update-packages uri used uri-path of <see cref="UpdateConfigurationFileUri"/> . 
+        ///     If <see cref="UpdateManager.UseDynamicUpdateUri"/> is false, return the original value.
+        /// </returns>
+        /// <exception cref="System.ArgumentException">updatePackageUri or literalVersion is a valid uri.</exception>
+        /// <exception cref="System.InvalidOperationException">updatePackageUri.AbsoluteUri is a valid uri.</exception>
+        public Uri ConvertPackageUri(Uri updatePackageUri, string literalVersion)
+        {
+            if (!UseDynamicUpdateUri)
+                return updatePackageUri;
+            if (updatePackageUri == null)
+                throw new ArgumentException("\'null\' is not valid for \'UpdatePackageUri\'", nameof(updatePackageUri));
```

Comment:
Use an ArgumentNullException only with the parameter name instead `throw new ArgumentNullException(nameof(updatePackageUri));`